PROVENANCE_SPEC.md

```diff
@@ -0,0 +1,193 @@
+# Provenance Spec
+
+## Model
+
+This provenance model has the following properties:
+* Each attestation has exaclty one `subject`, which is the same as an artifact or output of the build pipeline (e.g. OCI Images)
+* `materials` defines the inputs to the build system (e.g. Git repos)
+* `recipe` is the set of steps executed to build the `subject`, typically each container run by the TaskRun is a step in the `recipe`
+
+In this model, there can be multiple `materials` and multiple steps in the `recipe`. 
+However, there is exactly one `subject`.
+If a build pipeline outputs multiple artifacts, each will have its own attestation.
+
+![Diagram](images/provenance-diagram.png)
+
+
+## Schema
+
+```json
+{
+  "_type": "",
+  "name": "",
+  "predicateType": "",
+  "subject": {
+    "name": "",
+    "digest": null
+  },
+  "predicate": {
+    "invocation": {
+      "parameters": null,
+      "uri": "",
+      "event_id": ""
+    },
+    "recipe": {
+      "steps": [
+        {
+          "entryPoint": "",
+          "arguments": null,
+          "environment": null,
+          "annotations": null
+        }
+      ]
+    },
+    "metadata": {
+      "buildStartedOn": null,
+      "buildFinishedOn": null,
+      "reproducible": false
+    },
+    "materials": [
+      {
+        "uri": "",
+        "digest": null
+      }
+    ]
+  }
+}
+```
+
+## Fields
+
+`subject`, required, `object`
```

Comment:
I think we decided we might want to make subject repeated. Still not entirely sure...